pkg/metadata/container_test.go

```diff
@@ -0,0 +1,180 @@
+	"time"
+
+	assertlib "github.com/stretchr/testify/assert"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/metadata/store"
+
+	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+func TestContainerState(t *testing.T) {
+	for c, test := range map[string]struct {
+		metadata *ContainerMetadata
+		state    runtime.ContainerState
+	}{
+		"unknown state": {
+			metadata: &ContainerMetadata{
+				ID:   "1",
+				Name: "Container-1",
+			},
+			state: runtime.ContainerState_CONTAINER_UNKNOWN,
+		},
+		"created state": {
+			metadata: &ContainerMetadata{
+				ID:        "2",
+				Name:      "Container-2",
+				CreatedAt: time.Now().UnixNano(),
+			},
+			state: runtime.ContainerState_CONTAINER_CREATED,
+		},
+		"running state": {
+			metadata: &ContainerMetadata{
+				ID:        "3",
+				Name:      "Container-3",
+				CreatedAt: time.Now().UnixNano(),
+				StartedAt: time.Now().UnixNano(),
+			},
+			state: runtime.ContainerState_CONTAINER_RUNNING,
+		},
+		"exited state": {
+			metadata: &ContainerMetadata{
+				ID:         "3",
+				Name:       "Container-3",
+				CreatedAt:  time.Now().UnixNano(),
+				FinishedAt: time.Now().UnixNano(),
+			},
+			state: runtime.ContainerState_CONTAINER_EXITED,
+		},
+	} {
+		t.Logf("TestCase %q", c)
+		assertlib.Equal(t, test.state, test.metadata.State())
+	}
+}
+
+func TestContainerStore(t *testing.T) {
+	containers := map[string]*ContainerMetadata{
+		"1": {
+			ID:        "1",
+			Name:      "Container-1",
+			SandboxID: "Sandbox-1",
+			Config: &runtime.ContainerConfig{
+				Metadata: &runtime.ContainerMetadata{
+					Name:    "TestPod-1",
+					Attempt: 1,
+				},
+			},
+			ImageRef:   "TestImage-1",
+			Pid:        1,
+			CreatedAt:  time.Now().UnixNano(),
+			StartedAt:  time.Now().UnixNano(),
+			FinishedAt: time.Now().UnixNano(),
+			ExitCode:   1,
+			Reason:     "TestReason-1",
+			Message:    "TestMessage-1",
+		},
+		"2": {
+			ID:        "2",
+			Name:      "Container-2",
+			SandboxID: "Sandbox-2",
+			Config: &runtime.ContainerConfig{
+				Metadata: &runtime.ContainerMetadata{
+					Name:    "TestPod-2",
+					Attempt: 2,
+				},
+			},
+			ImageRef:   "TestImage-2",
+			Pid:        2,
+			CreatedAt:  time.Now().UnixNano(),
+			StartedAt:  time.Now().UnixNano(),
+			FinishedAt: time.Now().UnixNano(),
+			ExitCode:   2,
+			Reason:     "TestReason-2",
+			Message:    "TestMessage-2",
+		},
+		"3": {
+			ID:        "3",
+			Name:      "Container-3",
+			SandboxID: "Sandbox-3",
+			Config: &runtime.ContainerConfig{
+				Metadata: &runtime.ContainerMetadata{
+					Name:    "TestPod-3",
+					Attempt: 3,
+				},
+			},
+			ImageRef:   "TestImage-3",
+			Pid:        3,
+			CreatedAt:  time.Now().UnixNano(),
+			StartedAt:  time.Now().UnixNano(),
+			FinishedAt: time.Now().UnixNano(),
+			ExitCode:   3,
+			Reason:     "TestReason-3",
+			Message:    "TestMessage-3",
+			Removing:   true,
+		},
+	}
+	assert := assertlib.New(t)
+
+	c := NewContainerStore(store.NewMetadataStore())
+
```

Comment:
Not too much, in fact. But it feels weird that we only have test for one metadata store.

After https://github.com/containerd/containerd/pull/859 lands, this part will be refactored, we may only need in memory data structure for `sandbox` and `image`.

The code and test will be different then. :)